src/scripts/controllers/editClientDateOfBirthController.ts

```diff
@@ -0,0 +1,104 @@
+}
+
+/**
+ * Renders the edit client date of birth form for a given case reference.
+ 
+/**
+ * Renders the edit client date of birth form for a given case reference.
+ * @param {Request} req - Express request object
+ * @param {Response} res - Express response object
+ * @param {NextFunction} next - Express next middleware function
+ */
+export async function getEditClientDateOfBirth(req: Request, res: Response, next: NextFunction): Promise<void> {
+  const caseReference = safeString(req.params.caseReference);
+  
+  try {
+    const response = await apiService.getClientDetails(req.axiosMiddleware, caseReference);
+    
+    let formData = { day: '', month: '', year: '' };
+    let originalData = { day: '', month: '', year: '' };
+    
+    if (response.status === 'success' && response.data !== null) {
+      const dateOfBirth = safeString(response.data.dateOfBirth);
+      
+      if (dateOfBirth !== '') {
+        const parsedDate = parseDateString(dateOfBirth);
+        formData = parsedDate;
+        originalData = parsedDate;
+      }
+    }
+    
+    res.render('case_details/edit-date-of-birth.njk', {
+      caseReference,
+      formData,
+      originalData,
+      errorState: { hasErrors: false, errors: [] },
+      csrfToken: (req as RequestWithCSRF).csrfToken?.(),
+    });
+  } catch (error) {
+    next(error);
+  }
+}
+
+/**
+ * Handles POST request for editing client date of birth form.
+ * Validates input and either displays errors or updates the client details.
+ * @param {Request} req - Express request object
+ * @param {Response} res - Express response object  
+ * @param {NextFunction} next - Express next middleware function
+ */
+export async function postEditClientDateOfBirth(req: Request, res: Response, next: NextFunction): Promise<void> {
+  const caseReference = safeString(req.params.caseReference);
+  
+  try {
+    const validationErrors: Result<ValidationErrorData> = validationResult(req).formatWith(formatValidationError);
+    
+    if (!validationErrors.isEmpty()) {
+      // Handle validation errors by rendering the form with error messages
+      handleDateOfBirthValidationErrors(validationErrors, req, res, caseReference);
+      return;
+    }
+
+    // No validation errors - construct date and save to data service
+    const day = hasProperty(req.body, 'dateOfBirth-day') ? safeString(req.body['dateOfBirth-day']) : '';
+    const month = hasProperty(req.body, 'dateOfBirth-month') ? safeString(req.body['dateOfBirth-month']) : '';
+    const year = hasProperty(req.body, 'dateOfBirth-year') ? safeString(req.body['dateOfBirth-year']) : '';
+    
+    // Construct ISO date string (YYYY-MM-DD) for API
+    let dateOfBirth = '';
+    if (day !== '' && month !== '' && year !== '') {
+      const paddedDay = day.padStart(DATE_PADDING_WIDTH, DATE_PADDING_CHAR);
+      const paddedMonth = month.padStart(DATE_PADDING_WIDTH, DATE_PADDING_CHAR);
+      dateOfBirth = `${year}-${paddedMonth}-${paddedDay}`;
```

Comment:
I think this is a good candidate for a helper function in `src/scripts/helpers/dateFormatter.ts`. Then we can reuse this in the schema as well.

```typescript
function constructDateString(day: string, month: string, year: string): string {
  const paddedMonth = month.padStart(DATE_PADDING_WIDTH, DATE_PADDING_CHAR);
  const paddedDay = day.padStart(DATE_PADDING_WIDTH, DATE_PADDING_CHAR);
  return `${year}-${paddedMonth}-${paddedDay}`;
}
```